### PR TITLE
Introduce new Buy action

### PR DESCRIPTION
**Pull request by ipdae**
- based on #464 
- validate & transfer item using Order data
- record `OrderReceipt` to state.

### CODE REVIEW

File: Lib9c/Model/Mail/OrderBuyerMail.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using Bencodex.Types;
+using Nekoyume.Model.State;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Model.Mail
+{
+    [Serializable]
+    public class OrderBuyerMail : Mail
+    {
+        public readonly Guid OrderId;
```
Comment (hskim881028) [inline Lib9c/Model/Mail/OrderBuyerMail.cs:13]: `public override MailType MailType => MailType.Auction;`
Please override the mail type

File: Lib9c/Model/Mail/OrderSellerMail.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using Bencodex.Types;
+using Nekoyume.Model.State;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Model.Mail
+{
+    [Serializable]
+    public class OrderSellerMail : Mail
+    {
+        public readonly Guid OrderId;
```
Comment (hskim881028) [inline Lib9c/Model/Mail/OrderSellerMail.cs:13]: `public override MailType MailType => MailType.Auction;`
Please override the mail type

File: Lib9c/Model/Mail/OrderExpirationMail.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 namespace Nekoyume.Model.Mail
 {
+    [Serializable]
     public class OrderExpirationMail : Mail
     {
         public readonly Guid OrderId;
```
Comment (hskim881028) [inline Lib9c/Model/Mail/OrderExpirationMail.cs:13]: `public override MailType MailType => MailType.Auction;`
Please override the mail type

File: Lib9c/Model/Mail/CancelOrderMail.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+using Bencodex.Types;
+using Nekoyume.Model.State;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Model.Mail
+{
+    [Serializable]
+    public class CancelOrderMail : Mail
+    {
+        public readonly Guid OrderId;
```
Comment (hskim881028) [inline Lib9c/Model/Mail/CancelOrderMail.cs:13]: `public override MailType MailType => MailType.Auction;`
Please override the mail type

**Review by boscohyun — APPROVED** [review]

**Review by hskim881028 — APPROVED** [review]

**Pull request merged**